src/hotspot/share/opto/subnode.cpp

```diff
@@ -1623,6 +1623,23 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
     return new BoolNode( ncmp, _test.negate() );
   }
 
+  // Change "bool eq/ne (cmp (and (urshift X 4) 1) 0)" into "bool ne/eq (cmp (and X 8) 0)".
```

Comment:
If I compare the two:
```
bool eq/ne (cmp (and (urshift X 4) 1) 0)
bool ne/eq (cmp (and X 8) 0)
```
Then I see that the outer part is the same for both:
`bool ne/eq (cmp (...) 0)`

So why would this not just be a optimization for the pattern `AndI(Shift X shift, mask)` -> `AndI(X, new_mask)`?